mariukhna_lab3_var12.cpp

```diff
@@ -0,0 +1,298 @@
+#include <iostream>
+#include <string>
+#include <vector>
+
+using namespace std;
+
+class Product
+{
+public:
+    Product();
+    Product(int quantity);
+    Product(int quantity, int productCode);
+    Product(int quantity, int productCode, double price);
+    Product(int quantity, int productCode, double price, string supplier);
+    Product(int quantity, int productCode, double price, string supplier, string countryOrigin);
+    Product(int quantity, int productCode, double price, string supplier, string countryOrigin, string productName);
+    ~Product();
+
+    void SetQuantity(int quantity);
+    int GetQuantity();
+
+    void SetProductCode(int productCode);
+    int GetProductCode();
+
+    void SetPrice(double price);
+    double GetPrice();
+
+    void SetSupplier(const string &supplier);
+    const string &GetSupplier();
+
+    void SetProductName(const string &productName);
+    const string &GetProductName();
+
+    void SetCountryOrigin(const string &countryOrigin);
+    const string &GetCountryOrigin();
+
+    void ShowInformationProduct();
+    friend void StartWorkWithMassive(vector<Product> &products);
+    void ComparePrice(const Product &other);
+    void FindExpensiveProduct(const vector<Product> &other);
+
+private:
+    int quantity;
+    int productCode;
+    double price;
+    string supplier;
+    string productName;
+    string countryOrigin;
+};
+
+void Product::SetQuantity(int quantity)
+{
+    this->quantity = quantity;
+}
+
+int Product::GetQuantity()
+{
+    return quantity;
+}
+
+void Product::SetProductCode(int productCode)
+{
+    this->productCode = productCode;
+}
+
+int Product::GetProductCode()
+{
+    return productCode;
+}
+
+void Product::SetPrice(double price)
+{
+    this->price = price;
+}
+
+double Product::GetPrice()
+{
+    return price;
+}
+
+void Product::SetSupplier(const string &supplier)
+{
+    this->supplier = supplier;
+}
+
+const string &Product::GetSupplier()
+{
+    return supplier;
+}
+
+void Product::SetProductName(const string &productName)
+{
+    this->productName = productName;
+}
+
+const string &Product::GetProductName()
+{
+    return productName;
+}
+
+void Product::SetCountryOrigin(const string &countryOrigin)
+{
+    this->countryOrigin = countryOrigin;
+}
+
+const string &Product::GetCountryOrigin()
+{
+    return countryOrigin;
+}
+
+Product::Product()
+{
+    cout << "The default constructor is called" << endl;
+
+    this->quantity = 0;
+    this->productCode = 0;
+    this->price = 0;
+    this->supplier = "Not specified";
+    this->productName = "Not specified";
+    this->countryOrigin = "Not specified";
+}
+
+Product::Product(int quantity) : productCode(0), price(0), supplier("Not specified"), productName("Not specified"), countryOrigin("Not specified")
```

Comment:
it looks like you misunderstood the task about constructors.
let's discuss it in the class